### PR TITLE
[codex] Refactor execution queue docs for faster dispatch

### PR DESCRIPTION
**Pull request by Un3qual**
## Summary

- Add a `Next Curation Target` lane so an empty `Ready Now` queue still has bounded promotion candidates.
- Add blocker unblocker contracts and move long Done history into an archive to keep `queue.md` focused as the dispatcher.
- Tighten queue/docs validation so idle queue states require curation candidates and impl candidates name concrete non-doc target paths.
- Add active handoffs to umbrella plans that previously forced broad historical-plan scanning.

## Validation

- `bash scripts/check-execution-queue.sh`
- `bash scripts/check-docs.sh` (passes with the existing outside-Nix Prettier warning)
- `python3 -B -m py_compile scripts/check-execution-queue.py`
- `git diff --cached --check`

---
## Summary by cubic
Refactored the execution queue for faster, bounded dispatch with a curated promotion lane and stricter contract validation. When `Ready Now` is empty, dispatch uses `Next Curation Target` plus anchored blocker contracts; completion history moves to `docs/execution/done-archive.md`.

- **New Features**
  - Added a `Next Curation Target` lane with a strict schema (`blocked_id`, `candidate_child_id`, `kind`, `source_contract`, `why_next`, `target_paths`, `verification`, `promotion_check`).
  - Introduced `docs/execution/blocker-contracts.md` with a Promotion Contract Template; curation entries and plans now reference specific contract anchors.
  - Moved Done history to `docs/execution/done-archive.md`; `scripts/check-docs.sh` now requires both files and the template header.
  - Updated `docs/execution/README.md` and prompts to forbid repo-wide `docs/plans/**` scans when `Ready Now` is empty; use only `Next Curation Target`.
  - Hardened `scripts/check-execution-queue.py` with contract-anchor checks, stricter non-doc target-path rules for `impl`, and expanded edge-case tests.

- **Migration**
  - When `Ready Now` is empty, promote exactly one candidate from `Next Curation Target` using its linked contract, or replace the candidate with a better bounded one.
  - Ensure each blocked row has a matching contract section; link plans to the contract anchor and `docs/execution/queue.md` in `target_paths`.
  - Move completed items to `docs/execution/done-archive.md`, then prune; run `bash scripts/check-execution-queue.sh` and `bash scripts/check-docs.sh`.

<sup>Written for commit 2b1bdbd3d2534156e806eb76955fd138be134eb0. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Un3qual/jazz/pull/88?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

___

## **CodeAnt-AI Description**
**Add a bounded promotion lane for empty execution queues and move old completion history out of the dispatcher**

### What Changed
- When `Ready Now` is empty, the queue now uses a new `Next Curation Target` section to hold 1-3 ordered promotion candidates instead of forcing a broad docs scan
- Each blocked item now has a matching blocker contract section that spells out the next narrow child, target files, verification, and what stays out of scope
- Old completed rows were moved from `queue.md` into `done-archive.md` so the queue stays focused on current work
- Queue and plan checks now require the new blocker-contract and archive files, plus the new curation-lane fields and stricter target-path rules

### Impact
`✅ Faster queue dispatch when Ready Now is empty`
`✅ Fewer broad docs scans to find the next task`
`✅ Clearer handoff from blocked work to the next promotable child`
<details><summary><strong>💡 Usage Guide</strong></summary>

### Checking Your Pull Request
Every time you make a pull request, our system automatically looks through it. We check for security issues, mistakes in how you're setting up your infrastructure, and common code problems. We do this to make sure your changes are solid and won't cause any trouble later.

### Talking to CodeAnt AI
Got a question or need a hand with something in your pull request? You can easily get in touch with CodeAnt AI right here. Just type the following in a comment on your pull request, and replace "Your question here" with whatever you want to ask:
<pre>
<code>@codeant-ai ask: Your question here</code>
</pre>
This lets you have a chat with CodeAnt AI about your pull request, making it easier to understand and improve your code.

#### Example
<pre>
<code>@codeant-ai ask: Can you suggest a safer alternative to storing this secret?</code>
</pre>

### Preserve Org Learnings with CodeAnt
You can record team preferences so CodeAnt AI applies them in future reviews. Reply directly to the specific CodeAnt AI suggestion (in the same thread) and replace "Your feedback here" with your input:
<pre>
<code>@codeant-ai: Your feedback here</code>
</pre>
This helps CodeAnt AI learn and adapt to your team's coding style and standards.

#### Example
<pre>
<code>@codeant-ai: Do not flag unused imports.</code>
</pre>

### Retrigger review
Ask CodeAnt AI to review the PR again, by typing:
<pre>
<code>@codeant-ai: review</code>
</pre>

### Check Your Repository Health
To analyze the health of your code repository, visit our dashboard at [https://app.codeant.ai](https://app.codeant.ai). This tool helps you identify potential issues and areas for improvement in your codebase, ensuring your repository maintains high standards of code health.

</details>

## Summary by CodeRabbit

* **Documentation**
  * Enhanced execution queue governance documentation with clearer contracts and validation rules.
  * Introduced blocker-contract definitions for promotion-ready handoffs and work boundaries.
  * Moved completed historical execution items to a dedicated archive.
  * Updated plan references to point to blocker contracts as source of truth.

* **Tests**
  * Strengthened validation with stricter queue-entry consistency checks and archive-reuse prevention.

### CODE REVIEW

File: scripts/check-execution-queue.py [hunk 1]
```diff
@@ -293,25 +304,58 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+    real_non_doc_paths: list[tuple[str, Path]] = []
+    for target_path in target_paths:
+        if not target_path or target_path == "-":
+            continue
+        target_path_obj = normalize_target_path(target_path)
+        if target_path_obj.is_absolute() or ".." in target_path_obj.parts:
+            fail(f"{row_context} names non-repo-relative target path: {target_path}")
+            continue
+        if target_path_obj == Path("."):
+            continue
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:566]: Fixed in `5fabfbb`: `.` now fails as a non-concrete target path, including non-impl curation rows. Added the `curation dot target regression` fixture and verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

File: scripts/check-execution-queue.py [hunk 2]
```diff
@@ -293,25 +304,58 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+        resolved_target_path = (ROOT / target_path_obj).resolve()
+        try:
+            resolved_repo_relative = resolved_target_path.relative_to(ROOT)
+        except ValueError:
+            fail(f"{row_context} names non-repo-relative target path: {target_path}")
+            continue
+        if not resolved_target_path.is_file():
+            fail(f"{row_context} names missing or non-file target path: {target_path}")
+            continue
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `5fabfbb`: target paths are now validated for repo-relative shape and existing-directory mistakes, but planned new files are allowed. `impl` rows/candidates still require at least one existing non-doc owner path. Added the `curation planned target regression` fixture and verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

File: scripts/check-execution-queue.py [hunk 3]
```diff
@@ -480,6 +532,70 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
...
+
+    for key in ("why_next", "promotion_check"):
+        if not normalize_text(row.get(key, "")):
+            fail(f"{row_context} is missing {key}")
+
+    source_contract_path = extract_markdown_link_path(
+        normalize_text(row.get("source_contract", "")), "source_contract"
+    )
+    if source_contract_path and not source_contract_path.is_file():
+        fail(f"{row_context} links to missing source_contract: {source_contract_path}")
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `5fabfbb`: `source_contract` parsing now preserves the fragment, validates it against `blocker-contracts.md` markdown anchors, and requires the resolved heading to match the row's `blocked_id`. Added the `curation missing anchor regression` fixture and verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

File: scripts/check-execution-queue.py [hunk 4]
```diff
@@ -480,6 +532,70 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
...
+    fail(f"{QUEUE_PATH} Next Curation Target must contain at most 3 candidates")
+
+if not ready_rows and not curation_rows:
+    fail(
+        f"{QUEUE_PATH} Ready Now is empty, so Next Curation Target must contain "
+        "1-3 promotion candidates"
+    )
+
+blocked_ids = {normalize_text(row.get("id", "")) for row in blocked_rows}
+seen_candidate_ids: set[str] = set()
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:807]: Fixed in `5fabfbb`: the validator now parses `blocker-contracts.md` headings and fails any `Blocked` row without a matching section. The current file already had the six sections called out in the summary; this change enforces that invariant going forward. Added the `blocked missing contract regression` fixture and verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

File: scripts/check-execution-queue.py [hunk 5]
```diff
@@ -480,6 +532,70 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
...
+    elif blocked_id not in blocked_ids:
+        fail(f"{row_context} references blocked_id that is not in Blocked: {blocked_id}")
+
+    if not candidate_child_id:
+        fail(f"{row_context} is missing candidate_child_id")
+    elif candidate_child_id in all_ids:
+        fail(
+            f"{row_context} candidate_child_id already exists in queue sections: "
+            f"{candidate_child_id}"
+        )
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:825]: Fixed in `5fabfbb`: curation `candidate_child_id` values are now checked against ids in `done-archive.md` as well as active queue sections. Added the `curation archive id reuse regression` fixture and verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

File: scripts/check-execution-queue.py [hunk 6]
```diff
@@ -293,25 +304,58 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
-        fail(f"{QUEUE_PATH} plan link escapes repository root: {cell}")
+        fail(f"{QUEUE_PATH} {label} link escapes repository root: {cell}")
         return None
-    if plan_path.suffix.lower() not in DOC_SUFFIXES:
-        fail(f"{QUEUE_PATH} plan link must point to a text plan file: {cell}")
+    if link_path.suffix.lower() not in DOC_SUFFIXES:
+        fail(f"{QUEUE_PATH} {label} link must point to a text file: {cell}")
         return None
-    return plan_path
+    return link_path
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `5fabfbb`: this is covered by the new `source_contract` fragment validation and the typo-anchor regression fixture. Verified with `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, and `bash scripts/check-docs.sh`.

**Comment by Un3qual** [comment]
Greptile/Diffray summary risk areas addressed in `5fabfbb`:

- `source_contract` links now validate the section anchor and require it to match the curation row's `blocked_id`.
- `Blocked` rows now have a reverse check against `blocker-contracts.md` headings.
- `candidate_child_id` values now also reject reuse of ids in `done-archive.md`.
- target-path validation now rejects `.` placeholders while allowing planned new files when an `impl` row/candidate still names at least one existing non-doc owner path.

The current `blocker-contracts.md` already contained the six sections named as missing in the summary; the new validator check locks that invariant going forward.

Verification: `python3 -B -m py_compile scripts/check-execution-queue.py`; `bash scripts/test-check-execution-queue.sh`; `bash scripts/check-execution-queue.sh`; `bash scripts/check-docs.sh`; `git diff --check`; `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 7]
```diff
@@ -480,6 +644,83 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
         seen_ids[row_id] = section_name
         all_ids.add(row_id)
 
+contract_headings, contract_anchors = (
+    extract_markdown_heading_index(BLOCKER_CONTRACTS_PATH)
+    if blocked_rows or curation_rows
+    else (set(), {})
+)
+archived_ids = extract_done_archive_ids() if curation_rows else set()
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `48480ce`: archived ids are now loaded whenever active queue ids exist, and any `Ready Now`/`Blocked`/`Done` id that already appears in `docs/execution/done-archive.md` fails validation. Added the `active archive id reuse regression` fixture; the existing curation archive-id regression still covers candidate-child reuse. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 8]
```diff
@@ -553,52 +794,9 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
...
-                continue
-            if not resolved_target_path.is_file():
-                fail(
-                    f"{QUEUE_PATH} Ready Now row {row_id} names missing or non-file "
-                    f"target path: "
-                    f"{target_path}"
-                )
+        validate_target_paths(
+            f"{QUEUE_PATH} Ready Now row {row_id}", target_paths, True
+        )
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:953]: Fixed in `48480ce`: `Ready Now` target paths are now validated for every `kind`, with the existing non-doc owner-path requirement still limited to `kind: impl`. This closes `.`/escaping-path holes for docs and coordination rows without blocking valid docs-only targets. Added the `ready docs dot target regression` fixture and tightened the README contract. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

**Comment by Un3qual** [comment]
Greptile/Diffray follow-up on the updated summary items in `48480ce`:

- Greptile issue 1: wrong-file `source_contract` validation now returns after the path error, avoiding the misleading follow-on anchor failure. Added the `source contract wrong file regression` fixture.
- Greptile issue 2: added the `idle queue without curation regression` fixture for empty `Ready Now` with no `Next Curation Target` candidates.
- Diffray validation-complexity/data-consistency risks: active queue ids are now checked against `done-archive.md`, curation candidate ids remain checked, and `Ready Now` target-path validation applies to every `kind`.
- Diffray workflow-disruption/migration-guide suggestions: `docs/execution/README.md` is the queue maintenance guide for this PR and now calls out concrete target paths plus active/archive id reuse. The 15 blocker sections are intentional and validator-backed; one section per blocked row is the maintainability mechanism, so no separate migration document is needed in this change.

Verification: `python3 -B -m py_compile scripts/check-execution-queue.py`; `bash scripts/test-check-execution-queue.sh`; `bash scripts/check-execution-queue.sh`; `bash scripts/check-docs.sh`; `git diff --check`; `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 9]
```diff
@@ -293,25 +315,158 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+            fail(f"{path} has duplicate markdown anchor #{anchor}")
+            continue
+        anchors[anchor] = heading
+        headings.add(heading)
+    return headings, anchors
+
+
+def extract_done_archive_ids() -> set[str]:
+    if not DONE_ARCHIVE_PATH.exists():
+        return set()
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `060988f`: standalone queue validation now requires `docs/execution/done-archive.md` whenever archived ids are used, so deleting or renaming it no longer silently disables active/candidate archive-id checks. Added the `missing done archive regression` fixture, and `bash scripts/check-docs.sh` now runs the validator regression harness. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 10]
```diff
@@ -293,25 +315,158 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+        if not match:
+            continue
+        heading = normalize_text(match.group(2).strip())
+        anchor = markdown_heading_anchor(heading)
+        if not anchor:
+            continue
+        if anchor in anchors and anchors[anchor] != heading:
+            fail(f"{path} has duplicate markdown anchor #{anchor}")
+            continue
+        anchors[anchor] = heading
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `060988f`: markdown heading indexing now rejects any repeated anchor, including identical repeated `### JN-*` headings, so blocker contracts cannot create ambiguous `source_contract` targets. Added the `duplicate contract anchor regression` fixture and documented the invariant in `docs/execution/README.md`. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 11]
```diff
@@ -480,6 +645,88 @@ def parse_frontmatter(path: Path) -> dict[str, object] | None:
...
+    else:
+        validate_target_paths(row_context, target_paths, row_kind == "impl")
+
+    if normalize_list_item(row.get("verification", "")) == "-":
+        verification_commands: list[str] = []
+    else:
+        verification_commands = split_inline_list(
+            row.get("verification", ""), ";", normalize_list_item
+        )
+    if not verification_commands:
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `060988f`: curation rows now use the same verification parser as `Ready Now`, so `-` is only valid as the whole-cell placeholder and mixed `-; command` values fail with the malformed sentinel error. Added the `curation verification mixed sentinel regression` fixture and documented the rule. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

**Comment by Un3qual** [comment]
Diffray follow-up in `060988f`: tightened the validation-complexity/data-consistency risk areas by requiring `done-archive.md` during standalone queue validation, rejecting duplicate blocker-contract anchors, and rejecting mixed `-` verification sentinels in curation rows. Added focused regressions for all three and wired `bash scripts/test-check-execution-queue.sh` into `bash scripts/check-docs.sh` so the existing docs gate runs the validator harness. I did not add a separate migration guide or generator script: `docs/execution/README.md` is already the transition/maintenance guide, `blocker-contracts.md` already carries the template, and pruning/generation tooling is not needed for this docs-only refactor yet. There is no `.github/workflows` directory in this checkout to update directly.

File: scripts/check-execution-queue.py [hunk 12]
```diff
@@ -293,25 +315,172 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+            continue
+        if resolved_target_path.exists() and not resolved_target_path.is_file():
+            fail(f"{row_context} names non-file target path: {target_path}")
+            continue
+        if resolved_target_path.is_file() and not is_doc_target_path(
+            resolved_repo_relative
+        ):
+            real_non_doc_paths.append((target_path, target_path_obj))
+
+    if require_non_doc and not real_non_doc_paths:
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:584]: Fixed in `4f7ba90`: `Ready Now` target validation now requires every named target path to resolve to an existing file, so stale or mistyped impl paths fail even when another owner path exists. Curation candidates still allow planned child files while requiring an existing non-doc owner path. Added the `ready missing target regression` fixture and updated `docs/execution/README.md`. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

File: scripts/check-execution-queue.py [hunk 13]
```diff
@@ -293,25 +315,172 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+
+
+def extract_done_archive_ids() -> set[str]:
+    text = read_text_file(DONE_ARCHIVE_PATH, "done archive")
+    if text is None:
+        return set()
+    return {
+        normalize_text(match.group(1))
+        for match in re.finditer(r"^\|\s*`([^`]+)`\s*\|", text, re.MULTILINE)
+    }
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py]: Fixed in `4f7ba90`: `extract_done_archive_ids()` now parses the `done-archive.md` Done table by its `id` column instead of matching only backticked first cells. That extracts both ``| `ID` |`` and `| ID |` rows while ignoring headers/separators and malformed table rows. Added the `plain archive id reuse regression` fixture. Verified with `python3 -B -m py_compile scripts/check-execution-queue.py`, `bash scripts/test-check-execution-queue.sh`, `bash scripts/check-execution-queue.sh`, `bash scripts/check-docs.sh`, `git diff --check`, and `git diff --cached --check`.

**Comment by Un3qual** [comment]
Diffray follow-up for `4f7ba90`:

- Tightened the remaining validator/data-consistency edges: `done-archive.md` IDs are now parsed from the Done table `id` column, so plain and backticked archive rows both protect against reuse.
- Restored stricter `Ready Now` execution-target validation: every Ready Now target path must resolve to an existing file, while `Next Curation Target` can still mention planned child files as long as an existing non-doc owner path anchors the implementation candidate.
- Added focused regressions for both cases and kept the harness wired through `bash scripts/check-docs.sh`.

No separate migration guide, init template, pruning script, or generator was added in this PR: the README remains the transition/maintenance guide, `blocker-contracts.md` carries the template, and there is still no `.github/workflows` directory to wire directly.

File: scripts/check-execution-queue.py [hunk 14]
```diff
@@ -293,25 +315,234 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+        cells = split_markdown_row(line)
+        if len(cells) != len(headers):
+            fail(
+                f"{DONE_ARCHIVE_PATH} section 'Done' row {row_index} has "
+                f"{len(cells)} cells; expected {len(headers)}: {line}"
+            )
+            continue
+        row_id = normalize_text(cells[id_index])
+        if row_id:
+            archive_ids.add(row_id)
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:507]: Fixed in 2b1bdbd. extract_done_archive_ids() now tracks seen ids while parsing the Done table and fails duplicate archived ids, including plain/backticked row forms. Added the duplicate archive id regression. Verified with python3 -B -m py_compile scripts/check-execution-queue.py, bash scripts/test-check-execution-queue.sh, bash scripts/check-execution-queue.sh, bash scripts/check-docs.sh, and git diff --check.

File: scripts/check-execution-queue.py [hunk 15]
```diff
@@ -293,25 +315,234 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+        return
+    contract_heading = contract_anchors.get(fragment)
+    if contract_heading is None:
+        fail(f"{row_context} source_contract anchor not found: #{fragment}")
+        return
+    if blocked_id and contract_heading != blocked_id:
+        fail(
+            f"{row_context} source_contract points to {contract_heading}, "
+            f"not blocked_id {blocked_id}"
+        )
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:541]: Fixed in 2b1bdbd. The validator now parses Candidate child entries from blocker-contracts.md and requires a curation row candidate_child_id to match the linked source_contract section; missing Candidate child data is also reported. Added the curation candidate mismatch regression. Verified with python3 -B -m py_compile scripts/check-execution-queue.py, bash scripts/test-check-execution-queue.sh, bash scripts/check-execution-queue.sh, bash scripts/check-docs.sh, and git diff --check.

File: scripts/check-execution-queue.py [hunk 16]
```diff
@@ -293,25 +315,234 @@ def parse_markdown_table(section_name: str) -> tuple[list[str], list[dict[str, s
...
+def validate_target_paths(
+    row_context: str,
+    target_paths: list[str],
+    require_non_doc: bool,
+    require_existing: bool,
+) -> None:
+    real_non_doc_paths: list[tuple[str, Path]] = []
+    for target_path in target_paths:
+        if not target_path or target_path == "-":
+            continue
```
[the remark this answers is not in the document]
Comment (Un3qual) [reply scripts/check-execution-queue.py:559]: Fixed in 2b1bdbd. validate_target_paths() now rejects - when it is mixed with real paths, so both Ready Now and Next Curation Target rows fail on malformed target_paths sentinels while whole-cell - remains the missing-target placeholder. Added ready and curation mixed target sentinel regressions. Verified with python3 -B -m py_compile scripts/check-execution-queue.py, bash scripts/test-check-execution-queue.sh, bash scripts/check-execution-queue.sh, bash scripts/check-docs.sh, and git diff --check.

**Comment by Un3qual** [comment]
Reviewed CodeRabbit pre-merge warning after 2b1bdbd. I did not add blanket docstrings: this repo has no Python docstring-coverage config (no pyproject.toml/setup.cfg/.flake8/CodeRabbit config in this checkout), and the PR has one Python validator script covered by scripts/test-check-execution-queue.sh plus the docs gate. Adding docstrings across helper functions would be non-functional churn relative to the review findings. The current functional bot findings are fixed in 2b1bdbd and verified with python3 -B -m py_compile scripts/check-execution-queue.py, bash scripts/test-check-execution-queue.sh, bash scripts/check-execution-queue.sh, bash scripts/check-docs.sh, and git diff --check.
